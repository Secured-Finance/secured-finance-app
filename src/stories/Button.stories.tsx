import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Button, IButton } from 'src/components/new/Button';

export default {
    title: 'Components/Button',
    component: Button,
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

const Template: Story<IButton> = args => <Button {...args}>Button</Button>;

export const Default = Template.bind({});
Default.args = {
    outline: false,
};

export const Outlined = Template.bind({});
Outlined.args = {
    outline: true,
};
