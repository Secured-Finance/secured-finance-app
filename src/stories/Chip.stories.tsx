import React from 'react';
import { Story, Meta } from '@storybook/react';
import cm from './index.module.scss';

import { ChipButton, IChipButton } from 'src/components/new/Chip/ChipButton';
import { ChipLabel, IChipLabel } from 'src/components/new/Chip/ChipLabel';

export default {
    title: 'Components/Chips',
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

export const ChipButtons: Story<IChipButton> = () => (
    <div className={cm.elementsContainer}>
        <ChipButton>Button</ChipButton>
        <ChipButton accent={'purple'}>Button</ChipButton>
        <ChipButton accent={'red'}>Button</ChipButton>
    </div>
);

export const ChipLabels: Story<IChipLabel> = () => (
    <div className={cm.elementsContainer}>
        <ChipLabel label={'Registered'} />
        <ChipLabel label={'Working'} accent={'green'} />
        <ChipLabel label={'Terminated'} accent={'red'} />
        <ChipLabel label={'Closed'} accent={'grey'} />
        <ChipLabel label={'Past Due'} accent={'turquoise'} />
        <ChipLabel label={'Due'} accent={'blue'} />
    </div>
);
