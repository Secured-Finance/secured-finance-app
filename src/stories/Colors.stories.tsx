import { Meta, Story } from '@storybook/react';

export default {
    title: 'Design System/Colors',
    parameters: {
        chromatic: { disableSnapshot: false },
    },
} as Meta;

export const Primary: Story = () => (
    <div className='inline-flex flex-col items-center justify-start rounded-3xl bg-white p-10'>
        <div className='flex flex-col items-start justify-start space-y-6'>
            <p className='text-lg font-semibold leading-10 text-black-80'>
                Primary
            </p>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-universeBlue' />
                <div className='h-10 w-10 rounded-lg bg-horizonBlue' />
                <div className='h-10 w-10 rounded-lg bg-starBlue' />
                <div className='h-10 w-10 rounded-lg bg-teal' />
                <div className='h-10 w-10 rounded-lg bg-green' />
                <div className='h-10 w-10 rounded-lg bg-orange' />
                <div className='h-10 w-10 rounded-lg bg-red' />
            </div>
            <div className='h-10 w-10 rounded-lg bg-purple' />
        </div>
    </div>
);

export const Secondary: Story = () => (
    <div className='inline-flex flex-col items-start justify-start space-y-6 rounded-3xl bg-white p-10'>
        <p className='text-lg font-semibold leading-10 text-black-80'>
            Secondary
        </p>
        <div className='inline-flex items-center justify-start space-x-6'>
            <div className='h-10 w-10 rounded-lg bg-galacticOrange' />
            <div className='h-10 w-10 rounded-lg bg-planetaryPurple' />
            <div className='h-10 w-10 rounded-lg bg-nebulaTeal' />
            <div className='h-10 w-10 rounded-lg bg-secondaryPurple ' />
            <div className='h-10 w-10 rounded-lg bg-moonGrey' />
        </div>
    </div>
);

export const Shades: Story = () => (
    <div className='inline-flex flex-col items-center justify-start rounded-3xl bg-white p-10'>
        <div className='flex flex-col items-start justify-start space-y-6'>
            <p className='text-lg font-semibold leading-10 text-black-80'>
                Shades
            </p>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-starBlue-80' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-60' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-40' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-20' />
                <div className='h-10 w-10 rounded-lg bg-starBlue-10' />
            </div>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg bg-black-90' />
                <div className='h-10 w-10 rounded-lg bg-black-80' />
                <div className='h-10 w-10 rounded-lg bg-black-70' />
                <div className='h-10 w-10 rounded-lg bg-black-60' />
                <div className='h-10 w-10 rounded-lg bg-black-50' />
                <div className='h-10 w-10 rounded-lg bg-black-40' />
                <div className='h-10 w-10 rounded-lg bg-black-30' />
                <div className='h-10 w-10 rounded-lg bg-black-20' />
                <div className='h-10 w-10 rounded-lg bg-black-10' />
            </div>
            <div className='inline-flex items-center justify-start space-x-6'>
                <div className='h-10 w-10 rounded-lg border-2 bg-white-90' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-80' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-70' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-60' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-50' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-40' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-30' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-20' />
                <div className='h-10 w-10 rounded-lg border-2 bg-white-10' />
            </div>
        </div>
    </div>
);
