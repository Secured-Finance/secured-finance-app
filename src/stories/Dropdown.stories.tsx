import React from 'react';
import { Story, Meta } from '@storybook/react';
import { Dropdown, IDropdown } from 'src/components/new/Dropdown';
import { filecoin, telegram } from 'src/components/new/icons';

export default {
    title: 'Components/Dropdown',
    component: Dropdown,
    argTypes: { onClick: { action: 'clicked' } },
} as Meta;

const Template: Story<IDropdown> = args => {
    const [value, setValue] = React.useState(args.value ?? '');

    return (
        <Dropdown
            {...args}
            onChange={e => {
                setValue(e.target.value);
            }}
            value={value}
        />
    );
};

let value = 1;
export const Default = Template.bind({});
Default.args = {
    options: [
        { value: 1, label: 'One' },
        { value: 2, label: 'Two', icon: filecoin },
        { value: 3, label: 'Three looooong' },
        { value: 4, label: 'Long with icon', icon: telegram },
    ],
    value,
    label: '',
    noBorders: false,
};
