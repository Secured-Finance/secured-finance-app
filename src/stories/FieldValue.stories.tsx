import React from 'react';
import { Story, Meta } from '@storybook/react';
import { FieldValue, IFieldValue } from 'src/components/new/FieldValue';

export default {
    title: 'Components/FieldValue',
    component: FieldValue,
} as Meta;

const Template: Story<IFieldValue> = args => <FieldValue {...args} />;

export const Default = Template.bind({});
Default.args = {
    field: '1Y Yield',
    value: '4.25%',
    small: false,
    bold: false,
};

export const Small = Template.bind({});
Small.args = {
    field: '1Y Yield',
    value: '4.25%',
    small: true,
};

export const Bold = Template.bind({});
Bold.args = {
    field: '1Y Yield',
    value: '4.25%',
    bold: true,
};
