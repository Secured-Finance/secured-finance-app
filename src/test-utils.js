import { render as rtlRender } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from 'src/store/reducers';

function render(
    ui,
    {
        initialState,
        store = createStore(reducer, initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>;
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
