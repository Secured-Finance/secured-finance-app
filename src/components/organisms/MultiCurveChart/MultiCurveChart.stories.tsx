import { RESPONSIVE_PARAMETERS, VIEWPORTS } from '.storybook/constants';
import type { Meta, StoryFn } from '@storybook/react';
import { Rate } from 'src/utils';
import { MultiCurveChart } from './MultiCurveChart';

export default {
    title: 'Organism/MultiCurveChart',
    component: MultiCurveChart,
    args: {
        title: 'Yield Curve Title',
        curves: {
            WFIL: [
                new Rate(9486),
                new Rate(8484),
                new Rate(7478),
                new Rate(6473),
                new Rate(4452),
                new Rate(9442),
                new Rate(9426),
                new Rate(9416),
            ],
            WBTC: [
                new Rate(11386),
                new Rate(12384),
                new Rate(13378),
                new Rate(14373),
                new Rate(15352),
                new Rate(16342),
                new Rate(17326),
                new Rate(18316),
            ],
            ETH: [
                new Rate(9486),
                new Rate(8484),
                new Rate(7478),
                new Rate(6473),
                new Rate(4452),
                new Rate(9442),
                new Rate(9426),
                new Rate(9416),
            ],
            USDC: [
                new Rate(11386),
                new Rate(12384),
                new Rate(13378),
                new Rate(14373),
                new Rate(15352),
                new Rate(16342),
                new Rate(17326),
                new Rate(18316),
            ],
        },
        labels: [
            'label1',
            'label2',
            'label3',
            'label4',
            'label5',
            'label6',
            'label7',
            'label8',
        ],
    },
    parameters: {
        ...RESPONSIVE_PARAMETERS,
        chromatic: {
            viewports: [VIEWPORTS.MOBILE, VIEWPORTS.TABLET],
        },
    },
} as Meta<typeof MultiCurveChart>;

const Template: StoryFn<typeof MultiCurveChart> = args => (
    <MultiCurveChart {...args} />
);

export const Default = Template.bind({});

export const NonOverLapping = Template.bind({});
NonOverLapping.args = {
    curves: {
        WFIL: [
            new Rate(9486),
            new Rate(8484),
            new Rate(7478),
            new Rate(6473),
            new Rate(9452),
            new Rate(9442),
            new Rate(9426),
            new Rate(9416),
        ],
        WBTC: [
            new Rate(19386),
            new Rate(19384),
            new Rate(19378),
            new Rate(14973),
            new Rate(19352),
            new Rate(16942),
            new Rate(19326),
            new Rate(12316),
        ],
        ETH: [
            new Rate(9486),
            new Rate(1484),
            new Rate(5478),
            new Rate(2473),
            new Rate(9452),
            new Rate(3442),
            new Rate(8426),
            new Rate(5416),
        ],
        USDC: [
            new Rate(11386),
            new Rate(12384),
            new Rate(19378),
            new Rate(14373),
            new Rate(15352),
            new Rate(16342),
            new Rate(17326),
            new Rate(18316),
        ],
    },
};
