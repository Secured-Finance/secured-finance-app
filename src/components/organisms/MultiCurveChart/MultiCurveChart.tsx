import {
    CategoryScale,
    Chart as ChartJS,
    ChartOptions,
    ChartTypeRegistry,
    LinearScale,
    LineElement,
    PointElement,
    Title,
    Tooltip,
    TooltipItem,
} from 'chart.js';
import classNames from 'classnames';
import { useRef, useState } from 'react';
import { Line } from 'react-chartjs-2';
import {
    crossHairPlugin,
    defaultDatasets,
} from 'src/components/molecules/LineChart/constants';
import { currencyMap, CurrencySymbol, Rate, toCurrencySymbol } from 'src/utils';

ChartJS.register(
    LinearScale,
    PointElement,
    LineElement,
    Title,
    CategoryScale,
    Tooltip,
    crossHairPlugin
);

export const options: ChartOptions<'line'> = {
    responsive: true,
    maintainAspectRatio: true,
    interaction: {
        mode: 'index',
        intersect: false,
    },
    layout: {
        padding: {
            top: 50,
        },
    },
    elements: {
        point: {
            hoverRadius: 6,
            hoverBorderColor: '#FFFFFF',
            hoverBorderWidth: 2,
            backgroundColor: '#5162FF',
        },
    },
    scales: {
        y: {
            beginAtZero: true,
            display: true,
            ticks: {
                callback: function (value: string | number) {
                    return value.toString() + '%';
                },
            },
            grid: {
                borderDash: [2],
                drawBorder: false,
                borderColor: 'rgba(255, 255, 255, 0.1)',
            },
        },

        x: {
            beginAtZero: true,
            ticks: {
                color: 'rgba(255, 255, 255, 0.6)',
                font: {
                    lineHeight: 2.0,
                },
                padding: 10,
            },
            grid: {
                display: false,
                borderColor: 'rgba(255, 255, 255, 0.1)',
            },
        },
    },
    hover: {
        intersect: false,
    },
    plugins: {
        tooltip: {
            yAlign: 'bottom',
            caretPadding: 16,
            backgroundColor: 'rgba(47, 50, 65, 1)',
            borderWidth: 1,
            borderColor: 'rgba(52, 56, 76, 1)',
            displayColors: false,
            cornerRadius: 10,
            padding: 8,
            callbacks: {
                label: (item: TooltipItem<keyof ChartTypeRegistry>) => {
                    return `${item.dataset?.label || ''} ${
                        item.formattedValue
                    }%`;
                },
                title: () => {
                    return '';
                },
            },
        },
    },
    events: ['click', 'mousemove'],
};

const getData = (
    curves: Record<CurrencySymbol, Rate[]>,
    currencies: CurrencySymbol[],
    labels: string[]
) => {
    return {
        labels: labels,
        datasets: Object.keys(curves)
            .filter(v => currencies.includes(v as CurrencySymbol))
            .map(key => {
                const ccy = toCurrencySymbol(key);
                if (ccy === undefined) {
                    throw new Error(`Invalid currency symbol: ${key}`);
                }

                return {
                    label: key,
                    data: curves[ccy].map(r => r.toNormalizedNumber()),
                    borderColor: currencyMap[ccy].chartColor,
                    backgroundColor: currencyMap[ccy].chartColor,
                    ...defaultDatasets,
                };
            }),
    };
};

const CurveChip = ({
    ccy,
    onClick,
}: {
    ccy: CurrencySymbol;
    active: boolean;
    onClick: (ccy: CurrencySymbol) => void;
}) => {
    const [active, setActive] = useState(true);
    return (
        <div key={ccy} className='flex flex-row items-center justify-center'>
            <button
                data-testid='curve-chip'
                style={{ backgroundColor: currencyMap[ccy].chartColor }}
                className={classNames(
                    `typography-body-small w-14 rounded-2xl py-3 pt-2 pb-[6px] font-secondary text-xs font-semibold uppercase text-white`,
                    {
                        'opacity-50': !active,
                    }
                )}
                onClick={() => {
                    setActive(!active);
                    onClick(ccy);
                }}
            >
                <p>{ccy}</p>
            </button>
        </div>
    );
};

export const MultiCurveChart = ({
    title,
    curves,
    labels,
}: {
    title: string;
    curves: Record<CurrencySymbol, Rate[]>;
    labels: string[];
}) => {
    const chartRef = useRef<ChartJS<'line'>>(null);
    const [activeCurrencies, setActiveCurrencies] = useState<
        Set<CurrencySymbol>
    >(new Set(Object.keys(curves) as CurrencySymbol[]));

    const handleCurrencyClick = (ccy: CurrencySymbol) => {
        if (activeCurrencies.has(ccy)) {
            setActiveCurrencies(prev => {
                const newSet = new Set(prev);
                newSet.delete(ccy);
                return newSet;
            });
        } else {
            setActiveCurrencies(prev => {
                const newSet = new Set(prev);
                newSet.add(ccy);
                return newSet;
            });
        }
    };

    return (
        <div className='box-border rounded-b-2xl border border-[#2D4064] bg-[#2D4064]/20 px-6 py-7'>
            <div className='flex flex-row justify-between pb-8'>
                <h1 className='typography-modal-title flex text-[20px] capitalize leading-6 text-white'>
                    {title}
                </h1>
                <div className='flex flex-row items-center justify-evenly gap-3'>
                    {Object.keys(curves).map(key => {
                        const ccy = toCurrencySymbol(key);
                        if (ccy === undefined) {
                            throw new Error(`Invalid currency symbol: ${key}`);
                        }
                        return (
                            <CurveChip
                                key={ccy}
                                ccy={ccy}
                                active={true}
                                onClick={handleCurrencyClick}
                            />
                        );
                    })}
                </div>
            </div>
            <Line
                className='rounded-2xl bg-black-20 p-5'
                data={getData(curves, Array.from(activeCurrencies), labels)}
                options={options}
                ref={chartRef}
                onClick={() => {}}
                data-chromatic='ignore'
            />
        </div>
    );
};
