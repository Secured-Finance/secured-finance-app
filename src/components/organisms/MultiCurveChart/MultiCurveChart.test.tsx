import { composeStories } from '@storybook/react';
import { queries, render, RenderResult, waitFor } from 'src/test-utils.js';
import * as stories from './MultiCurveChart.stories';

const { Default } = composeStories(stories);

describe('MultiCurveChart Component', () => {
    it('should render a MultiCurveChart', () => {
        render(<Default />);
    });

    it.skip('should render a match the snapshot', async () => {
        let ag: RenderResult<typeof queries, HTMLElement>;
        await waitFor(() => {
            ag = render(<Default />);
        }).then(() => {
            expect(
                ag.container
                    .querySelector('canvas')
                    ?.getContext('2d')
                    ?.__getEvents()
            ).toMatchSnapshot();
        });
    });

    it.skip('should deactivate a curve when clicking on a chip', async () => {
        let ag: RenderResult<typeof queries, HTMLElement>;
        await waitFor(() => {
            ag = render(<Default />);
        }).then(() => {
            ag.getByText('BTC').click();
            expect(
                ag.container
                    .querySelector('canvas')
                    ?.getContext('2d')
                    ?.__getEvents()
            ).toMatchSnapshot();
        });
    });
});
