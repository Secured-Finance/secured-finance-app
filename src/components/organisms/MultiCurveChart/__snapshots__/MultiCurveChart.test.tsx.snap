// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiCurveChart Component should deactivate a curve when clicking on a chip 1`] = `
[
  {
    "props": {
      "a": 1,
      "b": 0,
      "c": 0,
      "d": 1,
      "e": 0,
      "f": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "setTransform",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "value": "left",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  {
    "props": {
      "value": "middle",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  {
    "props": {
      "text": "FIL",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "text": "BTC",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "0%",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "2%",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "value": "left",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  {
    "props": {
      "value": "middle",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  {
    "props": {
      "text": "FIL",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "text": "BTC",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label1",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label2",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label3",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label4",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label5",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label6",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label7",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label8",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "save",
  },
  {
    "props": {
      "a": 1,
      "b": 0,
      "c": 0,
      "d": 1,
      "e": 0,
      "f": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "resetTransform",
  },
  {
    "props": {
      "height": 0,
      "width": 0,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {},
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "restore",
  },
]
`;

exports[`MultiCurveChart Component should render a match the snapshot 1`] = `
[
  {
    "props": {
      "a": 1,
      "b": 0,
      "c": 0,
      "d": 1,
      "e": 0,
      "f": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "setTransform",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "value": "left",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  {
    "props": {
      "value": "middle",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  {
    "props": {
      "text": "FIL",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "text": "BTC",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "0%",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "2%",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "value": "left",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  {
    "props": {
      "value": "middle",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  {
    "props": {
      "text": "FIL",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "text": "BTC",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label1",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label2",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label3",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label4",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label5",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label6",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label7",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  {
    "props": {
      "text": "label8",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "measureText",
  },
  {
    "props": {
      "value": "12px "Helvetica Neue", 'Helvetica', 'Arial', sans-serif",
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
]
`;
